Sources/Datadog/Core/System/AppStateListener.swift

```diff
@@ -0,0 +1,148 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Foundation
+import class UIKit.UIApplication
+
+/// A data structure to represent recorded app states in a given period of time
+internal struct AppStateHistory: Equatable {
+    /// Snapshot of the app state at `date`
+    struct Snapshot: Equatable, Comparable {
+        let isActive: Bool
+        let date: Date
+
+        static func < (lhs: Snapshot, rhs: Snapshot) -> Bool {
+            return lhs.date < rhs.date
+        }
+    }
+
+    var initialState: Snapshot
+    private(set) var changes = [Snapshot]()
+    // NOTE: RUMM-1064 changes.last.date > finalDate case isn't handled as not realistic
+    var finalDate: Date
+    var finalState: Snapshot {
+        return Snapshot(
+            isActive: (changes.last ?? initialState).isActive,
+            date: finalDate
+        )
+    }
+
+    mutating func add(change: Snapshot) {
+        changes.append(change)
+        changes.sort()
```

Comment:
Why does it require sorting? Is it possible that changes will be added in the random-time order?